src/main/java/com/nionios/trial/api/TalentUserEndpoint.java

```diff
@@ -20,20 +25,48 @@
     @Autowired
     private ExpenditureService expenditureService;
 
+    @Autowired
+    private TalentManagerService talentManagerService;
+
+    @Autowired
+    private TalentService talentService;
+
     @POST
-    @Path("/add/expenditure")
+    @Path("/{id}/add/expenditure")
     @Consumes(MediaType.APPLICATION_JSON)
     @Produces(MediaType.TEXT_PLAIN)
-    public Response addExpenditure(Expenditure expenditure) {
+    public Response addExpenditure(Expenditure expenditure, @PathParam("id") Long id) {
+        Talent talent = talentService.displayTalent(id);
         Expenditure result = talentUserService.saveExpenditure(expenditure);
+        List<Expenditure> talentExpenditures = talent.getExpenditures();
+        talentExpenditures.add(result);
+        talent.setExpenditures(talentExpenditures);
+        talentManagerService.addTalent(talent);
         return Response.accepted(result.getId()).build();
     }
 
     @DELETE
-    @Path("/delete/expenditure/{id}")
+    @Path("/{id}/delete/expenditure/{ex_id}")
     @Consumes(MediaType.APPLICATION_JSON)
-    public Response deleteExpenditure(@PathParam("id") Long id) {
-        Expenditure expenditure = expenditureService.findExpenditure(id);
+    public Response deleteExpenditure(@PathParam("id") Long id, @PathParam("ex_id") Long ex_id) {
```

Comment:
Rename ex_id to expenditure_id, I first thought it meant the former id of some object.